Implemented eslint with configuration file issue #4

helloworld-chhanda opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

- Implemented eslint with configuration file
-Fixed issue https://github.com/MAVRICK-1/e-commerce_website/issues/4

<!--- Describe your changes in detail -->

## Motivation and Context
I have implemented the eslint configuration but when I'm running the lint I can see around 25 files have linting errors. I have attached screenshot of the same which can be fixed in separate subsequent PRs. 

<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## How Has This Been Tested?
I have started the project and it is running properly and there is no change in actual project files.

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):
<img width="1440" alt="Screenshot 2024-05-12 at 3 23 46 PM" src="https://github.com/MAVRICK-1/e-commerce_website/assets/63163253/5f301c5c-55ea-40db-a568-c11b6559f981">

## Types of changes
- Added two scripts in package.json 
- added 4 new dev dependencies 
- added a new configuration file for eslint (.eslintrc.js)

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

MAVRICK-1 commented:
@ratnesh-maurya can you check the problem it's getting build error

ratnesh-maurya commented:
@MAVRICK-1  firstly we should not accept any direct commit to main branch and there should be a branch protection rule for default branch and  

-  Make a develop brach ( this is the branch where contributors should raise a pr) 
- and once we are satisfied with the changes will me merged main branch 
- As far regarding the error I will look onto in by midnight.
 Thanks

MAVRICK-1 commented:
@ratnesh-maurya That's a good point , but this was one my side projects , so i didn't introduce it .  If you can go ahead and make necessary changes when you got time that would be great .

helloworld-chhanda commented:
@MAVRICK-1 @ratnesh-maurya , Thank you for reviewing my PR.
So, I have fixed the build error by doing the following:
1. Fixed peer-dependency versioning issue of eslint
2. For build script, I am disabling the eslint as in the existing project I am seeing too many errors are coming so for now either we can bypass eslint for build and just keep it for commit time check or else we can fix all warnings and errors and then do a build time check as well.

I totally agree about keeping a separate `dev` branch for merging PRs and then moving dev => main whenever you guys feel comfortable.
Let me know when dev branch is created, I will change the target branch accordingly

MAVRICK-1 commented:
@helloworld-chhanda there were a lot of mistake you did like not mentioning the react version in eslint file and not installing the correct version . Anyways I appreciate that you tried .
